operators/cmd/manager/main.go

```diff
@@ -234,12 +241,35 @@ func execute() {
 		log.Error(err, "invalid roles specified")
 		os.Exit(1)
 	}
-	operatorNamespace := viper.GetString(OperatorNamespaceFlag)
-	log.Info("Setting up controller", "roles", roles)
+
+	// Setup a Kubernetes client without cache to request resources in the operator namespace and not in the
+	// managed namespaces until we upgrade the controller-runtime to be able to read in n namespaces with the
+	// cache enabled
+	operatorClient, err := k8s.NewClientGo(cfg)
+	if err != nil {
+		log.Error(err, "unable to set operator k8s client")
+		os.Exit(1)
+	}
+
+	// Setup operator info
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		log.Error(err, "unable to create k8s clientset")
+		os.Exit(1)
```

Comment:
I'm a bit confused we are creating 2 different clients here, can't we reuse operatorClient instead of the new clientset?

Reply:
I may have missed something, I do not see a way to get the server version using the `operatorClient` (vs `clientset.Discovery().ServerVersion()`).

The `operatorClient client.Client` only implements the `restclient.Interface` of the `discovery.DiscoveryInterface` not the `discovery.ServerVersionInterface`.

What we can do is replace the `operatorClient` by the `clientset` by using `clientset.CoreV1().ConfigMaps(ns)`.

```diff
diff --git a/operators/pkg/about/info.go b/operators/pkg/about/info.go
index ff5a45c2..af52c0dc 100644
--- a/operators/pkg/about/info.go
+++ b/operators/pkg/about/info.go
@@ -75,14 +73,11 @@
 // getOperatorUUID returns the operator UUID by retrieving a config map or creating it if it does not exist.
-func getOperatorUUID(operatorClient client.Client, operatorNs string) (types.UID, error) {
-	c := k8s.WrapClient(operatorClient)
+func getOperatorUUID(clientset kubernetes.Interface, operatorNs string) (types.UID, error) {
+	c := clientset.CoreV1().ConfigMaps(operatorNs)
+
 	// get the config map
-	var reconciledCfgMap corev1.ConfigMap
-	err := c.Get(types.NamespacedName{
-		Namespace: operatorNs,
-		Name:      UUIDCfgMapName,
-	}, &reconciledCfgMap)
+	reconciledCfgMap, err := c.Get(UUIDCfgMapName, metav1.GetOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return types.UID(""), err
 	}
@@ -99,7 +94,7 @@ func getOperatorUUID(operatorClient client.Client, operatorNs string) (types.UID
 				UUIDCfgMapKey: string(newUUID),
 			},
 		}
-		err = c.Create(&cfgMap)
+		_, err = c.Create(&cfgMap)
 		if err != nil {
 			return types.UID(""), err
 		}
@@ -115,7 +110,7 @@ func getOperatorUUID(operatorClient client.Client, operatorNs string) (types.UID
 			reconciledCfgMap.Data = map[string]string{}
 		}
 		reconciledCfgMap.Data[UUIDCfgMapKey] = string(newUUID)
-		err := c.Update(&reconciledCfgMap)
+		_, err := c.Update(reconciledCfgMap)
 		if err != nil && !apierrors.IsNotFound(err) {
 			return types.UID(""), err
 		}

```

Reply:
Covered by 4b62b2a.